addons/stock/models/stock_orderpoint.py

```diff
@@ -86,38 +105,72 @@ def _compute_allowed_location_ids(self):
+        ])
+        self.allowed_route_ids = route_by_product.ids
+
+    @api.depends('rule_ids', 'product_id.seller_ids', 'product_id.seller_ids.delay')
+    def _compute_json_popover(self):
         for orderpoint in self:
-            rules = orderpoint.product_id._get_rules_from_location(orderpoint.location_id)
-            lead_days, lead_days_description = rules._get_lead_days(orderpoint.product_id)
-            lead_days_date = fields.Date.today() + relativedelta.relativedelta(days=lead_days)
+            if not orderpoint.product_id or not orderpoint.location_id:
+                orderpoint.json_lead_days_popover = False
+                continue
+            dummy, lead_days_description = orderpoint.rule_ids._get_lead_days(orderpoint.product_id)
             orderpoint.json_lead_days_popover = dumps({
                 'title': _('Replenishment'),
                 'popoverTemplate': 'stock.leadDaysPopOver',
-                'lead_days_date': fields.Date.to_string(lead_days_date),
+                'lead_days_date': fields.Date.to_string(orderpoint.lead_days_date),
                 'lead_days_description': lead_days_description,
                 'today': fields.Date.to_string(fields.Date.today()),
+                'trigger': orderpoint.trigger,
+                'qty_forecast': orderpoint.qty_forecast,
+                'qty_to_order': orderpoint.qty_to_order,
+                'product_min_qty': orderpoint.product_min_qty,
+                'product_max_qty': orderpoint.product_max_qty,
+                'product_uom_name': orderpoint.product_uom_name,
+                'virtual': orderpoint.trigger == 'manual' and orderpoint.create_uid.id == SUPERUSER_ID,
             })
+
+    @api.depends('rule_ids', 'product_id.seller_ids', 'product_id.seller_ids.delay')
+    def _compute_lead_days(self):
+        for orderpoint in self:
+            if not orderpoint.product_id or not orderpoint.location_id:
+                orderpoint.lead_days_date = False
+                continue
+            lead_days, dummy = orderpoint.rule_ids._get_lead_days(orderpoint.product_id)
+            lead_days_date = fields.Date.today() + relativedelta.relativedelta(days=lead_days)
             orderpoint.lead_days_date = lead_days_date
 
-    def _quantity_in_progress(self):
-        """Return Quantities that are not yet in virtual stock but should be deduced from orderpoint rule
-        (example: purchases created from orderpoints)"""
-        return dict(self.mapped(lambda x: (x.id, 0.0)))
+    @api.depends('route_id', 'product_id', 'location_id', 'company_id', 'warehouse_id', 'product_id.route_ids')
+    def _compute_rules(self):
+        for orderpoint in self:
+            if not orderpoint.product_id or not orderpoint.location_id:
+                orderpoint.rule_ids = False
+                continue
+            orderpoint.rule_ids = orderpoint.product_id._get_rules_from_location(orderpoint.location_id, route_ids=orderpoint.route_id)
```

Comment:
can you try to get rid of _compute_rules and _compute_lead_days ? we used a json computed field (_compute_json_popover) especially to not add a lot of computed fields that are only useful in the popover

Reply:
ping

Reply:
_compute_lead_days calls
_compute_qty calls
_get_product_context calls
_compute_lead_days

Issue with compute dependence. Check master-rr-bug-compute-arm
https://pastebin.com/ii8pJ7tQ